packages/keyring-controller/src/KeyringController.ts

```diff
@@ -717,9 +778,15 @@ export class KeyringController extends BaseControllerV2<
    * Add qr hardware keyring.
    *
    * @returns The added keyring
+   * @throws If a QRKeyring builder is not provided
+   * when initializing the controller
    */
-  private async addQRKeyring(): Promise<QRKeyring> {
-    return this.#keyring.addNewKeyring(KeyringTypes.qr);
+  private async addQRKeyring(): Promise<Keyring<Json>> {
+    // we need to pass an empty array of accounts as options
+    // otherwise `eth-keyring-controller` will apply `{}` as default and
+    // pass it to `QRKeyring.deserialize` method, that will cause
+    // `QRKeyring.accounts` var to be set to `undefined`.
```

Comment:
This sounds like a bug in KeyringController.  This is a bug that we've had before, and fixed (see https://github.com/MetaMask/KeyringController/pull/136)

Reply:
Fixed here: https://github.com/MetaMask/KeyringController/pull/252